### ISSUE TITLE
train speed is very lower

### ISSUE THREAD

**Issue by Man1978-scd** [microsoft/LightGBM#5348]
When I use lightGBM on the server, the training speed is very slow, the CPU and memory are sufficient, the data volume is about 2w, even slower than my own computer, I don't know the reason, I hope there is an answer，if i use Dask,it will be better?

**Comment by jameslamb** [comment]
Thanks for using LightGBM!

Without a [minimal, reproducible example](https://stackoverflow.com/help/minimal-reproducible-example), maintainers cannot help much with what you're asking. You haven't told us what you mean by "very slow", and you haven't provided enough information for us to understand what "data volume is about 2w" means. You haven't even told us how you are using LightGBM (R? Python? CLI?) or the version you are using.

With the information you've provided, I can only give you a few general suggestions:

* subscribe to #5336 to be notified of changes in that speed-related conversation
* if you have access to a GPU, try building a GPU-enabled version of LightGBM (https://lightgbm.readthedocs.io/en/latest/GPU-Tutorial.html)
* review the suggestions in https://lightgbm.readthedocs.io/en/latest/Parameters-Tuning.html#for-faster-speed for reducing training time
* see https://lightgbm.readthedocs.io/en/latest/Parallel-Learning-Guide.html#dask to learn how to experiment with the Dask interface

**Comment by Man1978-scd** [comment]
@jameslamb , thank for answer my question

here is my operation


- python=3.8.11

- [lightgbm=3.3.2](url)

my fit params is 
```json
{
          'boosting_type': 'dart',
          'learning_rate': 0.05,
          'max_depth': 20,
          'bagging_fraction': 0.6,
          'feature_fraction': 0.8,
          'bagging_freq': 3,
          'num_leaves': 30,
          'metric': 'rmse',
          'num_iterations': 400,
          'num_threads': os.cpu_count(),
          'tree_learner': 'feature'
        }
```
my train dataset shape is (14583,96)
```python

model = LGBMRegressor(**gbm_params)
model.fit(x_train, y_train)
```
x_train is a type of pd.DataFrame
when I fit model in the linux (not with GPU),It takes about 3 minutes,
the cpu use info like this when the code in the fit step
```text
Tasks: 269 total,   3 running, 266 sleeping,   0 stopped,   0 zombie
%Cpu0  : 96.9 us,  0.0 sy,  0.0 ni,  2.8 id,  0.0 wa,  0.0 hi,  0.0 si,  0.3 st
%Cpu1  : 95.1 us,  0.4 sy,  0.0 ni,  4.6 id,  0.0 wa,  0.0 hi,  0.0 si,  0.0 st
%Cpu2  : 90.9 us,  0.3 sy,  0.0 ni,  8.4 id,  0.0 wa,  0.0 hi,  0.0 si,  0.3 st
%Cpu3  : 93.7 us,  0.7 sy,  0.0 ni,  5.6 id,  0.0 wa,  0.0 hi,  0.0 si,  0.0 st
%Cpu4  : 91.9 us,  0.4 sy,  0.0 ni,  7.7 id,  0.0 wa,  0.0 hi,  0.0 si,  0.0 st
%Cpu5  : 94.4 us,  0.0 sy,  0.0 ni,  5.6 id,  0.0 wa,  0.0 hi,  0.0 si,  0.0 st
%Cpu6  : 99.6 us,  0.0 sy,  0.0 ni,  0.0 id,  0.0 wa,  0.0 hi,  0.0 si,  0.4 st
%Cpu7  : 93.0 us,  0.3 sy,  0.0 ni,  6.3 id,  0.0 wa,  0.0 hi,  0.0 si,  0.3 st
%Cpu8  : 90.8 us,  0.0 sy,  0.0 ni,  9.2 id,  0.0 wa,  0.0 hi,  0.0 si,  0.0 st
%Cpu9  : 97.9 us,  0.0 sy,  0.0 ni,  1.7 id,  0.0 wa,  0.0 hi,  0.3 si,  0.0 st
%Cpu10 : 91.6 us,  0.7 sy,  0.0 ni,  7.7 id,  0.0 wa,  0.0 hi,  0.0 si,  0.0 st
%Cpu11 : 93.0 us,  0.0 sy,  0.0 ni,  6.6 id,  0.0 wa,  0.0 hi,  0.0 si,  0.3 st
%Cpu12 : 97.9 us,  0.0 sy,  0.0 ni,  2.1 id,  0.0 wa,  0.0 hi,  0.0 si,  0.0 st
%Cpu13 :100.0 us,  0.0 sy,  0.0 ni,  0.0 id,  0.0 wa,  0.0 hi,  0.0 si,  0.0 st
%Cpu14 : 94.1 us,  0.7 sy,  0.0 ni,  4.9 id,  0.0 wa,  0.0 hi,  0.0 si,  0.3 st
%Cpu15 : 93.7 us,  0.3 sy,  0.0 ni,  5.9 id,  0.0 wa,  0.0 hi,  0.0 si,  0.0 st
```
but when I fit model is my computer ,it just takes about 10 seconds, my computer is MAC M1，
Is there such a big gap between MAC and linux? I don't know the reason. Can you help me how to improve the training speed on the linux?

**Comment by guolinke** [comment]
for such a small dataset, the speed should be very fast. Did you try to build LightGBM from source in the Linux machine?

**Comment by Man1978-scd** [comment]
> @Man1978-scd did you find any reason for slow train speed?

Let me explain
This problem occurs when I use a Docker image
First I created a virtual environment on my local machine and installed Lightgbm. Then I added the virtual environment (Path: /anaconda/envs/) into the Docker image for use. Lightgbm is very slow when I use Docker containers to run my python programs
My guess is that while the program works, Lightgbm's python environment is a direct copy of it, and it doesn't have the ability to call the underlying C++ (gcc), causing it to run very slowly. Later I built the Dockerfile to create the image, and the program worked fine
